lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/betConsumer.go

```diff
@@ -0,0 +1,73 @@
+package rabbitmq
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+	"log"
+
+	"github.com/pkg/errors"
+
+)
+
+type BetConsumer struct {
+	channel Channel
+	config  ConsumerConfig
+}
+
+func NewBetConsumer(channel Channel, config ConsumerConfig) (*BetConsumer, error) {
+	_, err := channel.QueueDeclare(
+		config.Queue,
+		config.DeclareDurable,
+		config.DeclareAutoDelete,
+		config.DeclareExclusive,
+		config.DeclareNoWait,
+		config.DeclareArgs,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "bet consumer initialization failed")
+	}
+
+	return &BetConsumer{
+		channel: channel,
+		config:  config,
+	}, nil
+}
+
+// Consume consumes messages until the context is cancelled. An error will be returned if consuming
+// is not possible.
+func (c *BetConsumer) Consume(ctx context.Context) (<-chan models.Bet, error) {
+	msgs, err := c.channel.Consume(
+		c.config.Queue,
+		c.config.ConsumerName,
+		c.config.AutoAck,
+		c.config.Exclusive,
+		c.config.NoLocal,
+		c.config.NoWait,
+		c.config.Args,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "bet consumer failed to consume messages")
+	}
+
+	bets := make(chan models.Bet)
+	go func() {
+		defer close(bets)
+		for msg := range msgs {
+			var bet models.Bet
+			err := json.Unmarshal(msg.Body, &bet)
```

Comment:
What happens if this fails?

Reply:
I can, however, propagate it, if that will be better.

Reply:
You resolved the issue I was aiming for. You added `continue` in the last PR :)

Before, you logged that error occurred and produced "gibberish" from `var bet models.Bet` into the `bets` channel.

So it works fine now. Please add `continue` to the `eventUpdateConsumer` as well.